samples/gocdk-blob/main_test.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2019 The Go Cloud Development Kit Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main_test
+
+import (
+	"flag"
+	"os"
+	"os/exec"
+	"testing"
+
+	"gocloud.dev/internal/testing/cmdtest"
+)
+
+var update = flag.Bool("update", false, "replace test file contents with output")
+
+func Test(t *testing.T) {
+	if err := exec.Command("go", "build").Run(); err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove("gocdk-blob")
+
+	ts, err := cmdtest.Read(".")
+	if err != nil {
+		t.Fatal(err)
+	}
+	ts.Commands["gocdk-blob"] = cmdtest.Program("gocdk-blob")
```

Comment:
By executing the binary this way, we lose any coverage data (look at the coverage comment on the PR -- it thinks coverage hasn't changed, when really it has). Is it possible to make the `gocdk-blob` hook call `main` instead of build the binary + execing it?

Reply:
Almost, but it's not pretty.

I just did it for blob. If you're okay with it, I'll move `execInProcess` to `cmdtest` for reuse.

Reply:
Yes, LGTM.

Reply:
Done, PTAL.